src/noc/containers/noc-field/utils/noc-validators.ts

```diff
@@ -1,188 +1,108 @@
-            const number = Number(timeValues[0]);
-            if (number > 23) {
-              return { datetime: true, hour: true, valid: false };
-            }
-          } else {
-            return { datetime: true, hour: true };
-          }
-          if (timeValues[1]) {
-            const number = Number(timeValues[1]);
-            if (number > 59) {
-              return { datetime: true, minute: true, valid: false };
-            }
-          } else {
-            return { datetime: true, minute: true };
-          }
-          if (timeValues[2]) {
-            const number = Number(timeValues[2]);
-            if (number > 59) {
-              return { datetime: true, second: true, valid: false };
-            }
-          } else {
-            return { datetime: true, second: true };
-          }
-        }
-      }
-      const momentValue = control.value.toString().replace('T', ' ').replace('.000', '');
-      if (!moment(momentValue, 'YYYY-MM-DD HH:mm:ss', true).isValid()) {
-        return { datetime: true };
+      const [year, month, day] = datePart.split('-').map(Number);
+      const [hour, minute, second] = (timePart || '').split(':').map(Number);
+
+      if (!isValidDateTime || year === 0 || month === 0 || month > 12 || day === 0 || day > 31 ||
+          hour > 23 || minute > 59 || (second !== undefined && second > 59)) {
+        return { month: true, datetime: true, valid: false };
       }
-      return;
+      return null;
     };
   }
 
   public static timeValidator(): ValidatorFn {
     return (control: AbstractControl): ValidationErrors | null => {
-      if (control.value === null || control.value === '') {
-        return;
+      if (this.isEmpty(control.value)) {
+        return null;
       }
-      const timeValues = control.value.toString().split(':');
-      if (timeValues) {
-        if (timeValues[0]) {
-          const number = Number(timeValues[0]);
-          if (number > 23) {
-            return { time: true, hour: true, valid: false, message: '' };
-          }
-        } else {
-          return { time: true, hour: true };
-        }
-        if (timeValues[1]) {
-          const number = Number(timeValues[1]);
-          if (number > 59) {
-            return { time: true, minute: true, valid: false };
-          }
-        } else {
-          return { time: true, minute: true };
-        }
-        if (timeValues[2]) {
-          const number = Number(timeValues[2]);
-          if (number > 59) {
-            return { time: true, second: true, valid: false };
-          }
-        } else {
-          return { time: true, second: true };
-        }
+      const value = control.value.toString();
+      const [hour, minute, second] = value.split(':').map(Number);
+      const isValidTime = moment(value, 'HH:mm:ss', true).isValid();
+      if (!isValidTime || hour > 23 || minute > 59 || (second !== undefined && second > 59)) {
+        return { hour: true, time: true, valid: false, message: '' };
```

Comment:
And again

Reply:
Please see my comments above. It's working as intended.